### ISSUE TITLE
Error using JSON editor with array of $ref objects

### ISSUE THREAD

**Issue by scottohara** [swagger-api/swagger-ui#2659]
One of my end users has reported a problem when using `type: "array"` in the schema of a request body parameter.

On closer inspection, the error only appears when the `jsonEditor` is set to `true`.

The error is:
`Uncaught TypeError: Cannot read property 'required' of undefined`, and occurs in the following line in `lib/jsoneditor.min.js`:

```
s.required && Array.isArray(s.required) && e.schema.required && Array.isArray(e.schema.required) && (n.required = e.schema.required.concat(s.required)));
```

(`s` is the undefined reference.  `s` is assigned a few lines above as `s = this.types[t]`; and the problem seems to be that `this.types` is an empty array - `[]`)

Below is a reduced YAML example that should demonstrate the issue.

1. Set `jsonEditor` to false and the API renders correctly
2. Set `jsonEditor` to true and the above error occurs

```yaml
swagger: "2.0"
info:
  title: "array test"
paths:
  /test:
    post:
      parameters:
      - in: "body"
        name: "FooList"
        required: true
        schema:
          $ref: "#/definitions/FooList"
      responses:
        200:
          description: "200 response"
definitions:
  Foo:
    type: "object"
    properties:
      unique_id:
        type: "string"
  FooList:
    type: "array"
    items:
      $ref: "#/definitions/Foo"
```

**Comment by zeConehead** [comment]
Just noticed the same error!

**Comment by iamsimakov** [comment]
The same problem. It's real cool feature. Guys, are you going to fix it? Thanks

**Comment by cabbonizio** [comment]
One of my teammates reported this last week and wanted to show another hand that we are affected also.

**Comment by iamsimakov** [comment]
it works if maximize `jsoneditor.min.js` and simplify ternary operator in 2583 line to view:
```
n = i({}, e.schema), n.type = s
// was:
// "string" == typeof s ? (n = i({}, e.schema), n.type = s) : 
// (n = i({}, e.schema, s), n = e.jsoneditor.expandRefs(n), s.required && Array.isArray(s.required) && e.schema.required && Array.isArray(e.schema.required) && (n.required = e.schema.required.concat(s.required)));
```
Bbut maybe it works some ugly. Anyway thanks for Swagger UI.

**Comment by scottohara** [comment]
It seems that this issue can be resolved by updating to the latest version ([v0.7.28](https://github.com/jdorn/json-editor/blob/master/dist/jsoneditor.min.js)) of the JSON editor. (Version currently used in this project is [v0.7.22](https://github.com/swagger-api/swagger-ui/blob/master/lib/jsoneditor.min.js))/

Pull request coming shortly....
